app/models/miq_provision_virt_workflow.rb

```diff
@@ -377,7 +377,7 @@ def allowed_provision_types(_options = {})
   def allowed_snapshots(_options = {})
     result = {}
     return result if (vm = get_source_vm).blank?
-    vm.snapshots.each { |ss| result[ss.id.to_s] = ss.current? ? "#{ss.name} (Active)" : ss.name }
+    vm&.snapshots&.each { |ss| result[ss.id.to_s] = ss.current? ? "#{ss.name} (Active)" : ss.name }
```

Comment:
the first ampersand:

`vm` can't be nil - if it were nil, then the previous line would have bailed out.

So this seems to be a check between using orchestration template (which does not implement `snapshots`) and vms (which do)

in this case, please use `try`

Keep the second ampersand the way it is. that protecting from the first `try` returning a nil.

Reply:
@kbrock Thanks for looking at this.
```
(byebug) vm&.snapshots222
nil
(byebug) vm
#<OpenStruct id=50, evm_object_class=:OrchestrationTemplate, name="test-clone-linked2">
```

If `vm` is an `Vm` object, what you said makes perfect sense.
But it is an `OpenStruct` hash which seems returns `nil` for any not defined attributes.

I will change the code as you suggested so it might be easier to understand for other people.

Reply:
thanks - yea, openstruct seems to implement any method asked.

I expected a try, but looks like ampersand works - up to you